src/ui/components/InitiativesEditor.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import React from "react";
+import {Admin, Resource, Loading} from "react-admin";
+import {
+  InitiativeList,
+  InitiativeCreate,
+  InitiativeEdit,
+} from "../components/Initiatives";
+import fakeDataProvider from "ra-data-fakerest";
+import {fakeInitiatives} from "../mock/fakeInitiatives";
+import {createMemoryHistory} from "history";
+import {createMuiTheme} from "@material-ui/core/styles";
+import pink from "@material-ui/core/colors/pink";
+import {load, type Props, type State} from "./ExplorerApp";
+
+const dataProvider = fakeDataProvider(fakeInitiatives, true);
+
+const history = createMemoryHistory();
+
+const theme = createMuiTheme({
+  palette: {
+    type: "dark",
+    primary: pink,
+  },
+});
+
+export default class App extends React.Component<Props, State> {
```

Comment:
naming nit:
```suggestion
export default class InitiativesEditor extends React.Component<Props, State> {
```

Reply:
Would be better to implement this as a functional component with hooks since class components are pretty much deprecated at this point:
```tsx
const InitiativesEditor = () => {
  const [loadResult, setLoadResult] = React.useState<LoadResult | null>(null);
  React.useEffect(() => {
    load().then(setLoadResult);
  }, []);

  if (!loadResult) {
    return (
      <Loading
        loadingPrimary="Fetching cred details..."
        loadingSecondary="Your patience is appreciated"
      />
    );
  }
  switch (loadResult.type) {
    case "FAILURE":
      return (
        <div>
          <h1>Load Failure</h1>
          <p>Check console for details.</p>
        </div>
      );
    case "SUCCESS":
      return (
        <Admin theme={theme} dataProvider={dataProvider} history={history}>
          <Resource
            name="initiatives"
            list={InitiativeList}
            create={InitiativeCreate(loadResult.credView)}
            edit={InitiativeEdit(loadResult.credView)}
          />
        </Admin>
      );
    default:
      throw new Error((loadResult.type: empty));
  }
};

export default InitiativesEditor;
```

Reply:
addressed this in #1968